tiledb/sm/cpp_api/query_condition_experimental.h

```diff
@@ -38,6 +38,100 @@
 namespace tiledb {
 class QueryConditionExperimental {
  public:
+  /* ********************************* */
+  /*          STATIC FUNCTIONS         */
+  /* ********************************* */
+
+  /**
+   * Factory function for creating a new set membership query condition.
+   *
+   * **Example:**
+   * @code{.cpp}
+   * tiledb::Context ctx;
+   * std::vector<int> values = {1, 2, 3, 4, 5};
+   * auto a1 = tiledb::QueryConditionExperimental::create(ctx, "a1", values);
+   * @endcode
+   *
+   * @param ctx The TileDB context.
+   * @param field_name The field name.
+   * @param values The set membership values to use.
+   */
+  template <typename T, impl::enable_trivial<T>* = nullptr>
+  static QueryCondition create(
+      const Context& ctx,
+      const std::string& field_name,
+      const std::vector<T>& values,
```

Comment:
nit: this could be a span, although it should be non-breaking to make that change in the future I think.